config/schedule.go

```diff
@@ -1,15 +1,82 @@
 package config
 
-import "time"
+import (
+	"time"
 
+	"github.com/creativeprojects/resticprofile/constants"
+)
+
+type ScheduleLockMode int8
+
+const (
+	// ScheduleLockModeDefault waits on acquiring a lock (local and repository) for up to ScheduleConfig lockWait (duration), before failing a schedule.
+	// With lockWait set to 0, ScheduleLockModeDefault and ScheduleLockModeFail behave the same.
+	ScheduleLockModeDefault = ScheduleLockMode(0)
+	// ScheduleLockModeFail fails immediately on a lock failure without waiting.
+	ScheduleLockModeFail = ScheduleLockMode(1)
+	// ScheduleLockModeIgnore does not create or fail on resticprofile locks. Repository locks cause an immediate failure.
+	ScheduleLockModeIgnore = ScheduleLockMode(2)
+)
+
+// Schedule is an intermediary object between the configuration (v1, v2+) and the ScheduleConfig object used by the scheduler.
+// The object is also used to display the scheduling configuration
 type Schedule struct {
-	Group      string        `mapstructure:"group"`
-	Profiles   []string      `mapstructure:"profiles"`
-	Command    string        `mapstructure:"run"`
-	Schedule   []string      `mapstructure:"schedule"`
-	Permission string        `mapstructure:"permission"`
-	Log        string        `mapstructure:"log"`
-	Priority   string        `mapstructure:"priority"`
-	LockMode   string        `mapstructure:"lock-mode"`
-	LockWait   time.Duration `mapstructure:"lock-wait"`
+	ProfileName             string            `mapstructure:"profile"`  // v1 only
```

Comment:
Can this be set by `mapstructure:"profile"`, a bit confusing when `profiles` can be set as well?
Could we use a single field `Profiles` instead and fill it with a the profile name?